### DISCUSSION TITLE
Various Issues (QXcbConnection)

### DISCUSSION THREAD

**Discussion by meteos77** [textbrowser/biblioteq#145] [General]
in my launch window I have this message : 
qt.qpa.xcb: QXcbConnection: XCB error: 3 (BadWindow), sequence: 1348, resource id: 10632152, major code: 40 (TranslateCoords), minor code: 0

this does not prevent it from working well but I prefer to point it out to you

**Comment by textbrowser** [comment]
Do these errors repeat?

**Comment by meteos77** [comment]
qt.qpa.xcb: QXcbConnection: XCB error: 3 (BadWindow), sequence: 1348, resource id: 10632152, major code: 40 (TranslateCoords), minor code: 0
qt.qpa.xcb: QXcbConnection: XCB error: 3 (BadWindow), sequence: 1628, resource id: 10633034, major code: 40 (TranslateCoords), minor code: 0
qt.qpa.xcb: QXcbConnection: XCB error: 3 (BadWindow), sequence: 4540, resource id: 10685635, major code: 40 (TranslateCoords), minor code: 0
qt.qpa.xcb: QXcbConnection: XCB error: 3 (BadWindow), sequence: 9172, resource id: 10698368, major code: 40 (TranslateCoords), minor code: 0
qt.qpa.xcb: QXcbConnection: XCB error: 3 (BadWindow), sequence: 10506, resource id: 10701696, major code: 40 (TranslateCoords), minor code: 0
I have not yet located the action that produces this but now I am watching

**Comment by meteos77** [comment]
it seems to come with the dialog box 
BiblioteQ User Error
trying to do an SRU or Z3950 search without an ISBN13

**Comment by textbrowser** [comment]
Did you see my comment about disabling logging?

**Comment by textbrowser** [comment]
https://doc.qt.io/qt-5/qloggingcategory.html

export QT_LOGGING_RULES='.debug=false;qt.qpa.=false'

**Reply by meteos77** [reply]
always the problem même avec la commande export

**Reply by meteos77** [reply]
I will have learned a new info about Qt :-)

**Comment by textbrowser** [comment]
QMessageBox::critical
	(this,
	 tr("BiblioteQ: User Error"),
	 tr("In order to query an Open Library site, either the ISBN-10 "
	    "or ISBN-13 must be provided."));
      QApplication::processEvents();
      id.id->setFocus();
      return;

Dialog is properly parented. Processing (https://doc.qt.io/qt-5/qcoreapplication.html#processEvents-1) events is fine. A set-focus? All appear normal.

I tried this several times and I do not see warnings in the terminal.

**Comment by textbrowser** [comment]
You're on Linux Mint which is favored for appearances, yes? Perhaps it's contributing to the warnings.

**Reply by meteos77** [reply]
so it's related to my machine, no matter

**Comment by textbrowser** [comment]
https://bugreports.qt.io/browse/QTBUG-87141

**Reply by meteos77** [reply]
status close ?

**Reply by textbrowser** [reply]
It's a non-threatening problem that is not easy to reproduce.

**Reply by meteos77** [reply]
subject closed
I would test with your compilation but with version 2021-10-10, the problem is also present on my machine

**Reply by textbrowser** [reply]
Test with the most current source and determine if the problem exists. Make sure you re-enable logging.

**Reply by textbrowser** [reply]
Are you retired?

**Reply by meteos77** [reply]
no

**Reply by meteos77** [reply]
the message also happens if I want to delete a user without selecting one

the message also happens if I want to delete a document without selecting one

**Reply by textbrowser** [reply]
> no

So you volunteer just because? :)

**Reply by meteos77** [reply]
the message arrives when I click on the OK button of the user error window

**Reply by meteos77** [reply]
for you there are only 2 possibilities retirement or active, I took a 3rd :-)

**Reply by textbrowser** [reply]
I'm not retired but I wish I was. I don't like working in capitalism.

**Reply by meteos77** [reply]
vast question, life leads us to paths and encounters that change our lives

**Reply by textbrowser** [reply]
Sure. Mostly in the negative.

**Reply by meteos77** [reply]
the disease makes you think a lot about your situation

**Reply by textbrowser** [reply]
Which disease? Capitalism? Indeed.

**Reply by meteos77** [reply]
but personal health is a point to be taken first

**Reply by meteos77** [reply]
I'm not retired but I wish I was, I had this thought while starting to work :-)

**Reply by meteos77** [reply]
standard of living per capital in europe
in french : https://www.insee.fr/fr/statistiques/fichier/4797646/FPS2020-F8-4.pdf

**Reply by textbrowser** [reply]
Free higher-education? Free healthcare?

**Reply by meteos77** [reply]
not free but paid for in large part by compulsory contributions the more you earn, the more you pay into the common fund (as a rule)

**Reply by textbrowser** [reply]
But it is free for the recipient, yes?

**Reply by meteos77** [reply]
health : very serious illness : Yes
education: yes except "high schools

**Comment by textbrowser** [comment]
In France, does labor news reach the citizenry as easily as the news of the state?

**Reply by meteos77** [reply]
did not understand the question (translator??)

**Reply by textbrowser** [reply]
So. The grand news outlets of France, do they cover the news of labor (strikes, etc.)?

**Comment by textbrowser** [comment]
Like here in the US, huge scandals involving Amazon and Kelloggs. The laborers know this but the public is mostly unaware because the media does not cover them.

**Reply by meteos77** [reply]
yes, but we also have a regional press for the local facts ; we are a small country 1000km x 800km

**Reply by meteos77** [reply]
the standard of living within Europe is very disparate
the standard of living within the same country is also different

price of an apartment in Paris : 10 275 € /m2
price of an apartment in Saint Etienne : 1 767€ / m2
distance between Paris and Saint Etienne : 500 Km

**Reply by meteos77** [reply]
I moved from the Paris area to the Saint Etienne area and it's the good life :-)

**Reply by meteos77** [reply]
the 3eme way

**Reply by textbrowser** [reply]
In capitalism, if you don't have a college degree you should be treated as poorly as possible. Hence the Amazon and Kelloggs scandals. It is of course not mentioned that a college degree in the US earns you a lifetime of debt. :) So, debt to earn a statement that you belong to respectable society. Without a degree? Scrub those toilets! :)

**Reply by meteos77** [reply]
I think that the person who cleans the toilets is one of the essential people in a company ... tried to go to work a week if the person does not do the job :-)

**Comment by meteos77** [comment]
I saw in the release notes  April 10, 2021.
    High DPI support.
I wonder about this feature of the screens in general when you think?

**Comment by textbrowser** [comment]
I looked at BQ in Windows using Qt 5.12 in a high-DPI device. The translation is correct. The same for the Linux version. The Linux version requires some environment variables to be set. I don't know about Mac.

**Comment by meteos77** [comment]
for info I found a tool to manipulate the marc format in command line other than yaz (it's in perl)
https://github.com/LibreCat/Catmandu-MARC

**Reply by meteos77** [reply]
I made a lot of progress with Catmandu
there are still some problems with ISBN-10 and ISBN-13 and some parsing problems
with LibreOffice, I rectify the ISBN-10 column and the import is done

[fix.zip](https://github.com/textbrowser/biblioteq/files/7780251/fix.zip)

with the following command the file 110000-file1-utf8.pan and my source file (unimarc)
catmandu convert MARC to CSV --fix fix/unimarc_to_BiblioteQ_import_csv-v0.1.fix < 110000-fichier1-utf8.pan > 110000-BQ.csv

**Comment by meteos77** [comment]
for info : BiblioteQ does not work on ReactOS 0.4.14
marginal system it's a test info

I have an error  Entry point not found
The entry point of the CancelIoEx procedure could not be located in the dynamic link library KERNEL32.dll.

**Comment by textbrowser** [comment]
A Windows derivative. I would build it from source. :)
